spec/controllers/oauth_tokens_controller_spec.rb

```diff
@@ -48,12 +48,9 @@
 
   describe "for an unconfirmed user" do
     let(:user) { create :user, :as_unconfirmed }
-    it "should return a conventional spec-compliant error response" do
+    it "should not return a conventional spec-compliant error response" do
       post :create, format: :json, params: default_params_for_strategy
-      expect( response.code ).to eq "400"
-      json = JSON.parse( response.body )
-      expect( json["error"] ).to eq "invalid_grant"
-      expect( json["error_description"] ).not_to be_blank
+      expect( response.code ).to eq "200"
```

Comment:
We still want to issue errors for unconfirmed users after the grace period, which ends on `User::EMAIL_CONFIRMATION_REQUIREMENT_DATETIME`, so we still want a test like this. Instead of inverting this test like you've done, I would change the describe block to something like "for an unconfirmed user after the grace period" and try to mock Time.now or `allowed_unconfirmed_grace_period?` so the test can assess what happens after the grace period expires.

Then I would make different describe block that includes the test you have here to test that unconfirmed users *within* the grace period do receive successful responses to their oauth token requests.

Reply:
I guess the decision was to not have a grace period at all, so this isn't necessary.